### PR TITLE
fix: cursor is moved when opening file through search interface

### PR DESCRIPTION
**Pull request by SeriousBug**
Fixes an issue where the cursor would get moved to the start of the document if the user opened the document through something like the search window, which is supposed to move the cursor to the matching position.

#2144

### CODE REVIEW

File: packages/plugin-core/src/WorkspaceWatcher.ts [hunk 1]
```diff
@@ -423,15 +425,44 @@ export class WorkspaceWatcher {
...
+        // Since we delayed, a new document could have opened. Make sure we're still in the document we expect
+        if (
+          VSCodeUtils.getActiveTextEditor()?.document.uri.fsPath === startFsPath
+        ) {
+          const { line, character } = editor.selection.active;
+          // Move the cursor, but only if it hasn't already been moved by VSCode, another extension, or a very quick user
+          if (line === 0 && character === 0) {
+            editor.selection = new Selection(position, position);
+          } else {
+            Logger.info({
```
Comment (jonathanyeung) [inline packages/plugin-core/src/WorkspaceWatcher.ts:454]: do we actually need these in the logs? maybe debug level or something lower than `info`?

**Comment by jonathanyeung** [comment]
apart from comments above, lgtm

**Comment by SeriousBug** [comment]
> can we add a test for this by manually overriding the position in a test and using https://sinonjs.org/releases/v1.17.7/fake-timers/ ?

That doesn't work unfortunately. If faketimer is set up, VSCode completely freezes. I implemented the tests by wrapping around `setTimeout` and simulating how the watcher works instead.

File: packages/common-all/src/utils.ts [hunk 2]
```diff
@@ -865,3 +865,19 @@ export function cleanName(name: string): string {
 export function normalizeUnixPath(fsPath: string): string {
   return path.posix.format(path.parse(fsPath));
 }
+
+/** Wrapper(s) for easier testing, to wrap functions where we don't want to mock the global function. */
+export class Wrap {
```
Comment (kevinslin) [inline packages/common-all/src/utils.ts:870]: can you add documentation for this in [[Cook|dendron://dendron.docs/dev.process.qa.test#cook]]?

feel free to merge afterward (and add to weekly)

**Review by kevinslin — APPROVED** [review]

**Pull request merged**